build_runner/lib/src/entrypoint/options.dart

```diff
@@ -191,9 +182,10 @@ class ServeOptions extends SharedOptions {
       }
     }
 
-    var outputMap = _parseOutputMap(argResults);
-    var buildDirs = _buildDirsFromOutputMap(outputMap)
-      ..addAll(serveTargets.map((t) => t.dir));
+    var buildDirs = _parseBuildDirs(argResults);
+    for (var target in serveTargets) {
+      buildDirs.add(BuildDirectory(target.dir));
```

Comment:
[nit] FWIW I preferred the old `..addAll(serveTargets.map)` since it's about collecting, not side effects.

We can revisit this though when we'll be able to do:

```dart
var buildDirs = {
    ..._parseBuildDirs(argResults),
    ...serveTargets.map((t) => BuildDirectory(t.dir)),
};
```